From f88cac8364d5312e29208018909827d2da4a0f87 Mon Sep 17 00:00:00 2001
From: Kuiying Wang <kuiying.wang@intel.com>
Date: Tue, 19 Feb 2019 16:52:51 +0800
Subject: [PATCH] Define post code interfaces for post code manager

It includes one method and 2 properties.
properties:
    - name: CurrentBootCycleIndex
      description: >
          It is used to indicate current boot cycle index.
    - name: MaxBootCycleNum
      description: >
          The max cached boot cycles for post code.
methods:
    - name: GetPostCodes
      description: >
          Method to get the cached post code for each boot cycle.
TestBy: bitbake build

Signeoff-by: Kuiying Wang <kuiying.wang@intel.com>
---
 .../State/Boot/PostCode.interface.yaml             | 30 ++++++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 xyz/openbmc_project/State/Boot/PostCode.interface.yaml

diff --git a/xyz/openbmc_project/State/Boot/PostCode.interface.yaml b/xyz/openbmc_project/State/Boot/PostCode.interface.yaml
new file mode 100644
index 0000000..711749d
--- /dev/null
+++ b/xyz/openbmc_project/State/Boot/PostCode.interface.yaml
@@ -0,0 +1,30 @@
+description: >
+    Monitor Post code coming and buffer all of them based on boot cycle
+    into file system.
+
+properties:
+    - name: CurrentBootCycleIndex
+      type: uint16
+      description: >
+          It is used to indicate current boot cycle index.
+    - name: MaxBootCycleNum
+      type: uint16
+      description: >
+          The max cached boot cycles for post code.
+methods:
+    - name: GetPostCodes
+      description: >
+          Method to get the cached post code for each boot cycle.
+      parameters:
+        - name: Index
+          type: uint16
+          description: >
+              Index indicates which boot cycle of post codes is requested.
+      returns:
+        - name: codes
+          type: array[uint64]
+          description: >
+            An array of post codes of one boot cycle.
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
+          - xyz.openbmc_project.Common.Error.InvalidArgument
-- 
2.16.2

